services/src/main/java/org/apache/druid/cli/QueryJettyServerInitializer.java

```diff
@@ -95,6 +107,19 @@ public void initialize(Server server, Injector injector)
       );
     }
 
+    if (querySchedulerConfig.getNumThreads() > 0
+        && querySchedulerConfig.getNumThreads() < serverConfig.getNumThreads()
+        && serverConfig.isEnableQueryRequestsQueuing()) {
+      // Add QoS filter for query requests so they don't take up more than querySchedulerConfig#numThreads
+      log.info("Enabling QoS Filter on query requests with limit [%d].", querySchedulerConfig.getNumThreads());
+      JettyBindings.QosFilterHolder filterHolder = new JettyBindings.QosFilterHolder(
+          new String[]{"/druid/v2/*"},
```

Comment:
does this pick up some extra stuff that isn't exactly queries? e.g. `BrokerQueryResource` and `ClientInfoResource`? Its probably not a problem since the goal is like protecting health check endpoints...

Reply:
It will pick. I can make it more fine-grained if needed. As you pointed out, the need is to protect health check endpoints.

Reply:
clarified the intent inside the comment.